packages/web-client/app/routes/card-pay/balances.ts

```diff
@@ -43,6 +44,26 @@ const BALANCES_PANEL = {
 };
 
 export default class CardPayBalancesRoute extends Route {
+  queryParams = {
+    flow: {
+      refreshModel: true,
+    },
+  };
+
+  beforeModel(transition: any) {
+    const { flow, workflowPersistenceId } = transition.to.queryParams;
+
+    if (flow && !workflowPersistenceId) {
+      transition.abort(); // https://github.com/emberjs/ember.js/issues/17118
+      this.transitionTo(transition.to.name, {
+        queryParams: {
+          flow: flow,
+          workflowPersistenceId: uuidv4(), // TODO: Something shorter for a prettier URL?
```

Comment:
`http://localhost:4200/card-pay/balances?flow=issue-prepaid-card&workflowPersistenceId=ea310aea-c724-4c14-adf3-b9dd51329304` is a pretty long URL. Should we use a shorter param name and shorter hash?

Reply:
How about `flow-id` for the query param for the and using [short-uuid](https://www.npmjs.com/package/short-uuid) for the ID?

Reply:
Good suggestion, will try that.